compose/material3/material3/src/commonMain/kotlin/androidx/compose/material3/SearchBar.kt

```diff
@@ -449,6 +456,14 @@ private fun SearchBarInputField(
             .fillMaxWidth()
             .focusRequester(focusRequester)
             .onFocusChanged { if (it.isFocused) onActiveChange(true) }
+            .onKeyEvent {
+                if (it.key == Key.Escape) {
```

Comment:
Better to check whether it's a KEY_PRESSED event; otherwise you could call onActiveChange twice.

Reply:
> otherwise you could call onActiveChange twice

The focus will be cleared after first hit, so it's ok to react to any event from `Esc`

Reply:
> The focus will be cleared after first hit, so it's ok to react to any event from Esc

It seems that it depends on how onActiveChange implemented by the user?

Reply:
https://github.com/JetBrains/compose-multiplatform-core/pull/811